cypress/support/commands.js

```diff
@@ -0,0 +1,8 @@
+/// <reference types="Cypress" />
+const crypto = require('crypto');
+
+Cypress.Commands.add('loginAs', user => {
+  const loginToken = crypto.randomBytes(12).toString('base64');
+  cy.task('associateLoginToken', {user, loginToken});
+  cy.setCookie('loginToken', loginToken);
```

Comment:
This is relatively straightforward once you understand all the moving parts, but I'm impressed how quickly you learned those moving parts. Nice work.